src/storage/buffer_manager.cpp

```diff
@@ -350,7 +350,7 @@ BufferHandle BufferManager::Pin(shared_ptr<BlockHandle> &handle) {
 void BufferManager::AddToEvictionQueue(shared_ptr<BlockHandle> &handle) {
 	D_ASSERT(handle->readers == 0);
 	handle->eviction_timestamp++;
-	PurgeQueue();
+	// PurgeQueue();
```

Comment:
This should be uncommented again

Reply:
This was actually deliberate but its related to the memory deallocation pile-up from the Fannie Mae stress test (everyone is trying to push and pull from the queue simultaneously, it can slow things down. But I can defer it to another PR.

Reply:
I added the PurgeQueue because if we have plenty of memory, and never evict blocks, our concurrent queue fills up with dead references. This is essentially a memory leak, which lead to the OOM reaper killing my process many times when I was implementing the sort code.

That being said, we should probably try to fine a way to reduce contention here, it shouldn't slow us down

Reply:
In principle the PurgeQueue does not need to be called continuously since it erases multiple blocks at once, only sometimes, perhaps e.g. every 100 blocks appended to the queue, or maybe only when the queue exceeds a certain size.

Reply:
We can revisit this at another time. I was trying to work on two problems at once and performance is really not the focus of this PR.